src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -1698,19 +1697,12 @@ private FeedbackSessionResultsBundle getFeedbackSessionResultsForUserInSectionBy
             }
         }
 
-        StudentAttributes student = null;
-        Set<String> studentsEmailInTeam = new HashSet<String>();
-        if (role == UserRole.STUDENT) {
-            student = studentsLogic.getStudentForEmail(courseId, userEmail);
-            List<StudentAttributes> studentsInTeam = studentsLogic
-                    .getStudentsForTeam(student.team, courseId);
-            for (StudentAttributes teammates : studentsInTeam) {
-                studentsEmailInTeam.add(teammates.email);
-            }
-        }
-        
+        StudentAttributes student = getStudent(courseId, userEmail, role);
+
+        Set<String> teammateEmails = getTeammateEmails(courseId, student);
+
         InstructorAttributes instructor = null;
-        if (role == UserRole.INSTRUCTOR) {
+        if (isUserInstructor(role)) {
```

Comment:
This looks very similar to the above for ```StudentAttributes``` where we check if it's a student then get student.. do you think it'll be a good idea to also abstract this? It looks like this particular thing is being done a few times in this method + the next few..